src/aws_encryption_sdk/internal/formatting/encryption_context.py

```diff
@@ -64,11 +70,14 @@ def _serialize_encryption_context_list(encryption_context_list, serialized_conte
         # we check to make sure that we return the right type of error message for an overly long key or value
         except struct.error as struct_error:
             message = str(struct_error)
+            # a key or value were too long and struct gave us the expected error
             if message == "'H' format requires 0 <= number <= 65535":
-                # the key or value were too large
+                raise SerializationError("Key or Value are too large.  Maximum length is 65535")
+            # a key or value were too long and struct's error message is different (maybe an update)
+            if _key_or_value_too_large(encryption_context_list):
                 raise SerializationError("Key or Value are too large.  Maximum length is 65535")
             # else we can just raise the struct error as it was (unknown)
-            raise struct_error
+            raise SerializationError("Unknown Error with struct pack.  Could not serialize key or value")
```

Comment:
Why not pass through the `struct_error` as well? Why squash that info?

Reply:
The error now contains two errors.  The message as above in a SerializationError and the struct error.